lib/Dialect/TTNN/Transforms/TTNNWorkarounds.cpp

```diff
@@ -360,6 +361,100 @@ class TTNNOperandsWorkaroundsRewriter
+
+  LogicalResult matchAndRewrite(ttnn::AllReduceOp op,
+                                PatternRewriter &rewriter) const override {
+    RankedTensorType inputType =
+        mlir::cast<RankedTensorType>(op.getInput().getType());
+    llvm::SmallVector<int64_t> inputTypeShape(inputType.getShape());
+    size_t scatter_dim = op.getScatterDim();
+    int32_t scatter_num = op.getScatterNum();
+    auto device = op.getDevice();
+    Location loc = op.getLoc();
+
+    // TODO(wooseoklee): Once it supports two dimensional tensor
+    // (https://github.com/tenstorrent/tt-metal/issues/15010), we can remove
+    // this workaround solution.
+    if (inputTypeShape.size() < 4) {
+      std::vector<int64_t> reshapedInputShape(4, 1);
+      for (size_t i = 0; i < inputTypeShape.size(); ++i) {
+        reshapedInputShape[i + inputTypeShape.size()] = inputTypeShape[i];
+      }
+
+      ArrayAttr reshapedInputShapeAttr =
+          rewriter.getI32ArrayAttr(std::vector<int32_t>(
+              reshapedInputShape.begin(), reshapedInputShape.end()));
+
+      auto reshapedInputType =
+          RankedTensorType::Builder(inputType).setShape(reshapedInputShape);
+
+      ttnn::ReshapeOp preReshapeOp = rewriter.create<ttnn::ReshapeOp>(
+          loc, Type(reshapedInputType), op.getInput(), reshapedInputShapeAttr);
+
+      scatter_dim = scatter_dim + (4 - inputTypeShape.size());
+
+      reshapedInputShape[scatter_dim] =
+          static_cast<int32_t>(reshapedInputShape[scatter_dim] / scatter_num);
+
+      auto scatteredInputType =
+          RankedTensorType::Builder(inputType).setShape(reshapedInputShape);
+
+      ttnn::ReduceScatterOp reduceScatterOp =
+          rewriter.create<ttnn::ReduceScatterOp>(
+              loc, Type(scatteredInputType), preReshapeOp.getResult(), device,
+              scatter_dim, op.getMathOp());
+
+      RankedTensorType outputType = mlir::cast<RankedTensorType>(op.getType());
+      SmallVector<int64_t> outputTypeShape(outputType.getShape());
+
+      std::vector<int64_t> reshapedOutputShape(4, 1);
+      for (size_t i = 0; i < outputTypeShape.size(); ++i) {
+        reshapedOutputShape[i + outputTypeShape.size()] = outputTypeShape[i];
+      }
+
+      auto reshapedOutputType =
+          RankedTensorType::Builder(outputType).setShape(reshapedOutputShape);
+
+      ttnn::AllGatherOp allGatherOp = rewriter.create<ttnn::AllGatherOp>(
+          loc, Type(reshapedOutputType), reduceScatterOp.getResult(), device,
+          scatter_dim);
+
+      ArrayAttr reshapedOutputShapeAttr = rewriter.getI32ArrayAttr(
+          std::vector<int32_t>(outputTypeShape.begin(), outputTypeShape.end()));
+
+      rewriter.replaceOpWithNewOp<ttnn::ReshapeOp>(op, Type(outputType),
+                                                   allGatherOp.getResult(),
+                                                   reshapedOutputShapeAttr);
+    } else {
+      // TODO(wooseoklee): Once all_reduce support in ttnn is stable, we can
```

Comment:
Can we also link an issue in a comment related to the metal side here?

Reply:
The issue is added in the comment. https://github.com/tenstorrent/tt-metal/issues/13835